util/validate/index.mjs

```diff
@@ -45,78 +45,71 @@ const run = ({
   const sharedSchemasFolder = path.join(__dirname, '..', '..', 'src', 'schemas')
   const externalFolder = path.join(__dirname, '..', '..', 'src', 'external')
   const modulesFolder = path.join(srcFolderArg, 'modules')
-  const distFolder = path.join('dist')
   
-  // Load all of the JSON-Schemas
+  const load = path => h.of(path)
+          .through(getSchemaContent)
+          .tap(addSchema)
+
+  const validate = (json, validator, prefix) => h(validator(json))
```

Comment:
Are any of the arguments to this function asynchronous? I can't figure out where `validator` comes from.

Reply:
validator is async, yes. it comes from the stream code in each of the if/else

Reply:
Yeah, I looked at it for a minute, but didn't immediately spot anything that was actually asynchronous. I see that you put `async` in front of it, but not sure why. Can revisit later. What I was getting at is that you have a `.flatMap(...).flatMap(...)` later in the code that maybe only needs to be `.flatMap(...).map(...)`. In other words you are doing async when you don't need to, maybe.